components/dashboard/application/environment/show.tsx

```diff
@@ -74,94 +63,61 @@ export const ShowEnvironment = ({ applicationId }: Props) => {
+				await refetch();
+			})
+			.catch(() => {
+				toast.error("Error to add build-args");
 			});
-		}
-	}, [form.reset, data, form, isEnvVisible]);
+	};
 
 	return (
 		<div className="flex w-full flex-col gap-5 ">
-			<Card className="bg-background">
-				<CardHeader className="flex flex-row w-full items-center justify-between">
-					<div>
-						<CardTitle className="text-xl">Environment Settings</CardTitle>
-						<CardDescription>
-							You can add environment variables to your resource.
-						</CardDescription>
-					</div>
-
-					<Toggle
-						aria-label="Toggle bold"
-						pressed={isEnvVisible}
-						onPressedChange={setIsEnvVisible}
-					>
-						{isEnvVisible ? (
-							<EyeOffIcon className="h-4 w-4 text-muted-foreground" />
-						) : (
-							<EyeIcon className="h-4 w-4 text-muted-foreground" />
-						)}
-					</Toggle>
-				</CardHeader>
-				<CardContent>
-					<Form {...form}>
-						<form
-							id="hook-form"
-							onSubmit={form.handleSubmit(onSubmit)}
-							className="w-full space-y-4"
-						>
-							<FormField
-								control={form.control}
-								name="environment"
-								render={({ field }) => (
-									<FormItem className="w-full">
-										<FormControl>
-											<CodeEditor
-												language="properties"
-												disabled={isEnvVisible}
-												placeholder={`NODE_ENV=production
-PORT=3000
-`}
-												className="h-96 font-mono"
-												{...field}
-											/>
-										</FormControl>
-
-										<FormMessage />
-									</FormItem>
-								)}
-							/>
-
-							<div className="flex flex-row justify-end">
-								<Button
-									disabled={isEnvVisible}
-									isLoading={isLoading}
-									className="w-fit"
-									type="submit"
-								>
-									Save
-								</Button>
-							</div>
-						</form>
-					</Form>
-				</CardContent>
-			</Card>
+			<Form {...envForm}>
+				<form onSubmit={envForm.handleSubmit(onEnvSubmit)}>
+					<Secrets
+						name="env"
+						isLoading={saveEnvironmentMutation.isLoading}
+						title="Environment Settings"
+						description="You can add environment variables to your resource."
+						placeholder={["NODE_ENV=production", "PORT=3000"].join("\n")}
+					/>
+				</form>
+			</Form>
+			{data?.buildType === "dockerfile" && (
+				<Form {...buildArgsForm}>
+					<form onSubmit={buildArgsForm.handleSubmit(onBuildArgsSubmit)}>
+						<Secrets
+							name="buildArgs"
+							isLoading={saveBuildArgsMutation.isLoading}
+							title="Build-time Variables"
+							description={
+								<span>
+									Available only at build-time. See documentation&nbsp;
+									<a
+										className="text-primary"
+										href="https://docs.docker.com/build/guide/build-args/"
+										target="_blank"
+										rel="noopener noreferrer"
+									>
+										here
+									</a>
+									.
+								</span>
+							}
+							placeholder="NPM_TOKEN=xyz"
+						/>
+					</form>
+				</Form>
+			)}
```

Comment:
We can safely revert this part to what we had, and just add a condition for dockerfile and we will just have 1 mutation and 1 form, the same as we had but just adding 1 more <FormField for the buildArgs

Reply:
I also considered using one form. The only thing that didn't convince me was that the save button outside the viewport, and you had to scroll to see it.

<img width="1327" alt="image" src="https://github.com/user-attachments/assets/a43dedf6-8710-430c-a371-772ce46c4f6e">

Should we try using flex-row instead of flex-column?

Reply:
Done! I don't think the save outside the viewport is a big deal. Let me know!